newspaper/mthreading.py

```diff
@@ -9,50 +9,49 @@
 __license__ = 'MIT'
 __copyright__ = 'Copyright 2014, Lucas Ou-Yang'
 
-import queue
-import traceback
-from threading import Thread
+from concurrent.futures import ThreadPoolExecutor, wait
+from threading import Lock
 
 from .configuration import Configuration
 
-
-class Worker(Thread):
-    """
-    Thread executing tasks from a given tasks queue.
-    """
-    def __init__(self, tasks, timeout_seconds):
-        Thread.__init__(self)
-        self.tasks = tasks
-        self.timeout = timeout_seconds
-        self.daemon = True
-        self.start()
-
-    def run(self):
-        while True:
-            try:
-                func, args, kargs = self.tasks.get(timeout=self.timeout)
-            except queue.Empty:
-                # Extra thread allocated, no job, exit gracefully
-                break
-            try:
-                func(*args, **kargs)
-            except Exception:
-                traceback.print_exc()
-
-            self.tasks.task_done()
-
+def perform_download_articles(paper):
+    return paper.download_articles
```

Comment:
This behavior has since been changed since your PR, can you replace this line with:

```
from newspaper.source import Source
...
if isinstance(paper, Source):
  return paper.download_articles
else:
  return paper.download
```